### PR TITLE
Feat/update to 2.6

### PR DESCRIPTION
**Pull request by Aquamars**
Here some changes:

1. Update google-cloud/pubsub version from 1.7.2 to 2.6.0
The breaking change should be not impact in Samsara.
https://github.com/googleapis/nodejs-pubsub/releases/tag/v2.0.0

2. Change subscription name to `${jobName}-{subscriptionName}`.
The subscription name should be shorter than previous version.

3. Auto create topic and subscription.
Auto create pubsub is convenient function.
CreateJob => auto create topic when not exist.
CeateWorker => auto create subscription and topic when not exist.

### CODE REVIEW

**Review by Calvin-Huang — APPROVED** [review]

**Comment by Calvin-Huang** [comment]
From our discussion before, we wanna unbind the topic - subscription name together for making topic/subscription name shorter. However, the new naming for the subscription combines job name with subscription name and it seems the duplicated name clips exist.

To make sure we're on the same page. I'm recognizing the new rule of `${jobName}-{subscriptionName}` would be that when we have `fetch-job` for the job name and `fetch-job` for the subscription name, the new name for the pubsub subscription would be `fetch-job-fetch-job`.

Can you give me some cases for the new naming to explain how the new naming works for shorter subscription name.

File: src/worker.js [hunk 1]
```diff
@@ -39,12 +48,17 @@ class Worker extends EventEmitter {
...
+      this.Subscription = await this.pubsub.createSubscriptionIfNotExists(
+        `${jobName}-${topicSuffix}`,
+        `${jobName}-${this.config.subscriptionName}`,
+      );
+    }
 
     const formattedSubscription = this.subscriber.subscriptionPath(
       this.config.projectId,
-      `${topicName}-${this.config.subscriptionName}`,
+      `${jobName}-${this.config.subscriptionName}`,
```
Comment (Aquamars) [inline src/worker.js:61]: > To make sure we're on the same page. I'm recognizing the new rule of ${jobName}-{subscriptionName} would be that when we have fetch-job for the job name and fetch-job for the subscription name, the new name for the pubsub subscription would be fetch-job-fetch-job.

Actually, the `this.config.subscriptionName` become the subscription suffix.

**Comment by Aquamars** [comment]
> Can you give me some cases for the new naming to explain how the new naming works for shorter subscription name.

Here are sample code of samsara:

```
export GOOGLE_PUBSUB_TOPIC_SUFFIX = pub
export GOOGLE_PUBSUB_SUBSCRIPTION_NAME = sub
```

```javascript
const samsara = require('samsara-queue');
const { createWorker, createJob }  = samsara(myCofig);

await createJob('jobName', myData).save();

const worker = createWorker();
const processing = async(data, done) => { console.log(data); await done();};
worker.process('jobName', processing, options);
```

The topic name is `jobName-pub`.
The subscription is `jobName-sub`.
Created by code.

File: src/worker.js (hunk 1 above)
Comment (Calvin-Huang) [reply src/worker.js:61]: 🤔

**Pull request merged**